pandas/core/dtypes/concat.py

```diff
@@ -58,6 +64,40 @@ def get_dtype_kinds(l):
     return typs
 
 
+def _cast_to_common_type(arr: ArrayLike, dtype: DtypeObj) -> ArrayLike:
```

Comment:
why is this private? everything else is not, so this seems odd

Reply:
This is an internal helper function for this file, which we indicate by using a leading underscore (the other functions here are used elsewhere in the pandas codebase)